contracts/hackatom/tests/integration.rs

```diff
@@ -186,3 +187,48 @@ fn failed_handle() {
         );
     });
 }
+
+#[test]
+fn handle_panic_and_loops() {
+    let mut deps = mock_instance(WASM);
+
+    // initialize the store
+    let verifier = HumanAddr(String::from("verifies"));
+    let beneficiary = HumanAddr(String::from("benefits"));
+    let creator = HumanAddr(String::from("creator"));
+
+    let init_msg = InitMsg {
+        verifier: verifier.clone(),
+        beneficiary: beneficiary.clone(),
+    };
+    let init_params = mock_params(
+        &deps.api,
+        creator.as_str(),
+        &coin("1000", "earth"),
+        &coin("1000", "earth"),
+    );
+    let init_res = init(&mut deps, init_params, init_msg).unwrap();
+    assert_eq!(0, init_res.messages.len());
+
+    // TRY PANIC
+    let handle_params = mock_params(&deps.api, beneficiary.as_str(), &[], &coin("1000", "earth"));
+    // panic inside contract should not panic out here
+    // Note: we need to use the production-call, not the testing call (which unwraps any vm error)
+    let handle_res = call_handle(
+        &mut deps,
+        &handle_params,
+        &to_vec(&HandleMsg::Panic {}).unwrap(),
+    );
+    assert!(handle_res.is_err());
+
+    // TRY INFINITE LOOP
+    // Note: we need to use the production-call, not the testing call (which unwraps any vm error)
+    deps.set_gas(1_000_000);
```

Comment:
This gas reset is not possible anymore in 0.7. Can you split the test to make upgrading easier?

Reply:
I just moved it up, we can set it at top - just there is some limit on the whole flow